node/src/transition_frontier/genesis/transition_frontier_genesis_config.rs

```diff
@@ -374,6 +363,22 @@ fn genesis_producer_stake_proof(mask: &ledger::Mask) -> v2::MinaBaseSparseLedger
     (&sparse_ledger).into()
 }
 
+#[derive(Debug, BinProtRead, BinProtWrite)]
+struct LedgerAccountsWithHash {
+    accounts: Vec<MinaBaseAccountBinableArgStableV2>,
```

Comment:
@Sventimir this doesn't include the merkle tree hashes, this means that this still requires re-hashing of every node in the tree to recompute it, which is the most expensive part of producing these ledgers.

The data stored should be the same the prebuilt config stores and rebuilt in the same way as `GenesisConfig::build_ledger_from_accounts_and_hashes` which loads the accounts and sets the hashes directly without any rehashing:

```rust
    ledger_hash: LedgerHash,
    accounts: Vec<MinaBaseAccountBinableArgStableV2>,
    hashes: Vec<(u64, LedgerHash)>,
```

Reply:
Done.

Reply:
small request, lets put `ledger_hash` as the first element, so that it can be read quickly without having to go through all the accounts first

Reply:
Done.